staging/src/k8s.io/apimachinery/pkg/util/sets/byte.go

```diff
@@ -18,10 +18,10 @@ package sets
 
 // Byte is a set of bytes, implemented via map[byte]struct{} for minimal memory consumption.
 //
-// Deprecated: use generic Set instead.
+// Deprecated: use generic OrderedSet instead.
```

Comment:
Why recommanding people migrate `Set` to `OrderedSet`? I don't believe that everyone cares about the order, and ordered listing could cost more performance through.

Reply:
Just for compatible with sets.String, sets.Int, there is no performance penalty if u do not call it.

Reply:
> Just for compatible with sets.String, sets.Int which has List() and UnsortedList() method, there is no performance penalty if u do not call it.

+1, Maybe we should have a better reason to make this change

Reply:
For now, if just use sets.Set[string] to replace sets.String,  `set.List()` should be replaced with `sets.List(set)`, it's so boring. And it should keep compatible .

https://github.com/kubernetes/client-go/blob/master/tools/cache/thread_safe_store.go#L339

or maybe `type Set[T comparable] map[T]Empty` should be replaced by `type Set[T cmp.Ordered] map[T]Empty`

Reply:
Greate point, and it could work with `buildin` type perfectly. After this merged, then maybe it time to initiate a migration to replace old typed set in k/k and other repos.

However now people can create a set which the type of key is struct. Replacing `comparable ` to `cmp.Ordered` may break the compatiblity. 

IMO, I like the original version more, which add a new ordered set and break nothing. My only concer is about the comment which recommend to use `OrderedSet`. Because only end-user know weather the set need order. I understand that there is additional no cost only without calling `List`. but it's strange and confusing to let people to use `OrderedSet` when they don't need order.

WDYT? @googs1025

Reply:
the code in string.go can be simplified. Or maybe we do not need to deprecate NewString.
```go
package sets

// String is a set of strings, implemented via map[string]struct{} for minimal memory consumption.
//
// Deprecated: use generic OrderedSet instead.
// new ways:
// s1 := OrderedSet[string]{}
// s2 := NewOrdered[string]()
type String = OrderedSet[string]

// NewString creates a String from a list of values.
func NewString(items ...string) String {
	return String(New[string](items...))
}

// StringKeySet creates a String from a keys of a map[string](? extends interface{}).
// If the value passed in is not actually a map, this will panic.
func StringKeySet[T any](theMap map[string]T) String {
	return String(KeySet(theMap))
}
```

Reply:
I'm not sure what the advantage of the change is, but I'm fine. Leave it to other reviewers or appover to decide. :)